src/rdf4cpp/rdf/datatypes/xsd/time/DateTimeStamp.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef RDF4CPP_DATETIMESTAMP_HPP
+#define RDF4CPP_DATETIMESTAMP_HPP
+
+#include <chrono>
+
+#include <rdf4cpp/rdf/datatypes/registry/DatatypeMapping.hpp>
+#include <rdf4cpp/rdf/datatypes/registry/FixedIdMappings.hpp>
+#include <rdf4cpp/rdf/datatypes/registry/LiteralDatatypeImpl.hpp>
+#include <rdf4cpp/rdf/datatypes/xsd/time/Timezone.hpp>
+
+namespace rdf4cpp::rdf::datatypes::registry {
+
+template<>
+struct DatatypeMapping<xsd_dateTimeStamp> {
+    using cpp_datatype = std::pair<TimeComparer::Timepoint, Timezone>;
+};
+
+
+template<>
+capabilities::Default<xsd_dateTimeStamp>::cpp_type capabilities::Default<xsd_dateTimeStamp>::from_string(std::string_view s);
+
+template<>
+std::string capabilities::Default<xsd_dateTimeStamp>::to_canonical_string(const cpp_type &value) noexcept;
+
+
+template<>
+std::partial_ordering capabilities::Comparable<xsd_dateTimeStamp>::compare(cpp_type const &lhs, cpp_type const &rhs) noexcept;
+
+extern template struct LiteralDatatypeImpl<xsd_dateTimeStamp,
+                                           capabilities::Comparable,
+                                           capabilities::FixedId>;
+
+}  // namespace rdf4cpp::rdf::datatypes::registry
+
+namespace rdf4cpp::rdf::datatypes::xsd {
+
+struct DateTimeStamp : registry::LiteralDatatypeImpl<registry::xsd_dateTimeStamp,
```

Comment:
Note for later: check if it makes sense to make this a subtype of `xsd:dateTime`. Regardless of whether xpath says so or not.

I would imagine being able to cast between them is helpful since they are so similar.